Automattic/wp-calypso
# SMP: Update tests for <SiteThumbnail>

#### Proposed Changes

#67029 has already merged, but I wanted to test that it correctly handled all the edge cases around retrying. I figured this'd be easier to do in a test environment since I can control the timing.

I'm not done with my test cases but I wanted to get this merged in case the migration to `testing-library` would make it easier to write tests that can exercise all the edge cases in the code that give rise to #67340

* Migrates tests to TypeScript
* Migrates tests from Enzyme to `testing-library`
* Adds a test that checks the retry handling is correctly tidied up when component unmounts

You'll notice that I had to add a little unit test-specific check to the production code (the `'a8cIsLoading` check). The only way we have to check whether mShots is still loading is to check image dimensions, but since jsdom doesn't have a rendering engine it's unable to simulate image dimensions. You also can't forcefully set a value for `naturalWidth` in the tests because jsdom has (accurately) emulated the read-only nature of that property. I don't think it's too bad, although there might be another way to do it.

#### Testing Instructions



* Tests should pass
* There's been only a small change to production code, but it's still worth sanity checking `/sites` to make sure images are loading correctly

#### Pre-merge Checklist
